src/AcceptanceTests/SagaDataStorage/When_storing_saga_with_custom_saga_attribute.cs

```diff
@@ -65,18 +65,19 @@ public CustomizedCollectionSaga(Context testContext)
 
                 protected override void ConfigureHowToFindSaga(SagaPropertyMapper<CustomizedCollectionSagaData> mapper)
                 {
-                    mapper.ConfigureMapping<StartSaga>(m => m.SomeId).ToSaga(d => d.Id);
+                    mapper.ConfigureMapping<StartSaga>(m => m.SomeId).ToSaga(d => d.SomeId);
                 }
 
                 public Task Handle(StartSaga message, IMessageHandlerContext context)
                 {
-                    testContext.SagaId = Data.Id;
+                    testContext.SagaId = ((IContainSagaData)Data).Id;
```

Comment:
is this casting necessary?

Reply:
It'd compile without it. The `Id` attribute on the class (ContainSagaData) is not seen by the IntelliSense because it is hidden, but on the interface it is visible.

Reply:
Updated